packages/core/bootstrap/src/lib/burst-limit/index.ts

```diff
@@ -13,37 +13,77 @@ import { AdapterError, logger } from '../external-adapter'
 export * as actions from './actions'
 export * as reducer from './reducer'
 
-const BURST_CAP_BUFFER = 0.9
+const SECOND_LIMIT_RETRIES = 10
+const MINUTE_LIMIT_WARMER_BUFFER = 0.9
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
+
+const availableSecondLimitCapacity = async (
+  store: Store<BurstLimitState>,
+  burstCapacity1s: number,
+) => {
+  for (let retry = SECOND_LIMIT_RETRIES; retry > 0; retry--) {
+    const state = store.getState()
+    const { requests }: { requests: RequestsState } = state
+
+    const observedRequestsInSecond = selectTotalNumberOfRequestsFor(requests, IntervalNames.SECOND)
+
+    if (observedRequestsInSecond > burstCapacity1s) {
+      logger.debug(
+        `Per Second Burst rate limit cap of ${burstCapacity1s} reached. ${observedRequestsInSecond} requests sent in the last minute. Waiting 1 second. Retry number: ${retry}`,
+      )
+      await delay(1000)
+    }
+    return true
```

Comment:
Should this be in the `if` block?

Reply:
No. This function is checking if we are above the limit on requests we sent in the last second.
If so (if block) delay a second and check again.
If not quit out by returning true.

Reply:
but then won't this mean the `for` loop never goes past the first iteration as we'll always return `true` in the first iteration?

Reply:
Ah shoot, you are right. Missing an `else` there.